aiosmtpd/controller.py

```diff
@@ -159,15 +151,16 @@ def start(self):
         ready_event.wait(self.ready_timeout)
         if self._thread_exception is not None:  # pragma: on-wsl
             # See comment about WSL1.0 in the _run() method
-            assert self._thread is not None  # Stupid LGTM.com; see github/codeql#4918
             raise self._thread_exception
+        if not ready_event.is_set():
+            raise TimeoutError("SMTP server failed to start within allotted time")
```

Comment:
Previously we just wait and didn't actually do anything if timeout is reached.

This is now fixed, inspired by #244